### ISSUE TITLE
Svelte 4 support

### ISSUE THREAD

**Issue by rallets** [svelteness/svelte-jester#121]
Hi, I'm wondering if this library is still maintained. 
Svelte 4 has been released few days ago, and I needed to change how the tests were written, to be able to get them to work. 
Is this something it's planned on your side?
Thanks

**Comment by sebastianrothe** [comment]
Hey, yes it is. I saw the v4 release and plan to have a look at what must be changed.

If you just want to test and dont rely on jest, you should consider vitest, which proves to be very good with ESM format.

**Comment by rallets** [comment]
Hi, glad to hear that. I just changed a bit here and there to make them work, nothing blocking on my side. So I don't see any value for me currently to switch to vitest, but thanks anyway on the tip, really appreciated.
FYI I had problems with
- running tests that depend on `svelte/store`
- running tests that import `Svelte components`.
The problem looks like is raised from how the code is transpiled/compiled in the new Svelte.
Basically I test just pure functions and it works well.

**Comment by tarun-khanna** [comment]
Hey @sebastianrothe, any update here on svelte v4 support ?
Facing similar issues mentioned by @rallets

**Comment by sebastianrothe** [comment]
Just reading through https://svelte.dev/docs/v4-migration-guide#removal-of-cjs-related-output means there will be no CJS version of svelte-jester. This new version will use ESM entirely.

**Comment by benmccann** [comment]
You'll need to use ESM with Svelte 4:
- Set `NODE_OPTIONS=--experimental-vm-modules`
- Set the `"extensionsToTreatAsEsm": ['.svelte', '.ts']` Jest config option
- And until a new version of `svelte-jester` is released, you'll have to manually include the `.mjs` extension:
```
transform: {
  "^.+\\.svelte$": [
    "./node_modules/svelte-jester/dist/transformer.mjs",
    {
      "preprocess": true
    }
  ]
},
```

If you're using `@testing/library-svelte` you'll also need to update it to the latest version.

You may also consider [migrating to vitest](https://vitest.dev/guide/migration.html), which generally works much better in my experience and is the officially recommended unit testing solution.

**Comment by sebastianrothe** [comment]
Released as v3.0.0 https://www.npmjs.com/package/svelte-jester
